draft-ietf-tls-tls13.md

```diff
@@ -2190,21 +2196,105 @@ identities
   extension (see {{early-data-indication}}), the first identity is the
   one used for 0-RTT data.
 
+binders
+: A series of HMAC values, one for
+  each PSK offered in the "pre_shared_keys" extension and in the same
+  order, computed as described below.
+
 selected_identity
 : The server's chosen identity expressed as a (0-based) index into
   the identities in the client's list.
 {: br}
 
+Each PSK is associated with a single Hash algorithm. For PSKs established
+via the ticket mechanism ({{NewSessionTicket}}), this is the Hash used for
+the KDF. For externally established PSKs, the Hash algorithm MUST be set when the
+PSK is established.
+
+Prior to accepting PSK key establishment, the server MUST validate the
+corresponding binder value (see {{psk-binder}} below) If this
+value is not present or does not validate the server MUST abort the
+handshake.  Servers SHOULD NOT attempt to validate multiple binders;
+rather they SHOULD select a single PSK and validate solely its binder.
+
+In order to accept PSK key establishment, the server sends a
+"pre_shared_key" extension with the selected identity. 
+
+Clients MUST verify that the server's selected_identity is within the
+range supplied by the client, that the same cipher suite was selected
+and that the, "key_share", and
+"signature_algorithms" extensions are consistent with the indicated
+ke_modes and auth_modes values. If these values are not consistent,
+the client MUST abort the handshake with an "illegal_parameter" alert.
+
+This extension MUST be the last extension in the ClientHello (this
+facilitates implementation as described below). Servers MUST check
+that it is the last extension and otherwise fail the handshake with an
+"illegal_parameter" alert.
+
+If the server supplies an "early_data" extension, the client MUST
+verify that the server selected the first offered identity. If any
+other value is returned, the client MUST abort the handshake
+with an "unknown_psk_identity" alert.
+
+#### PSK Binder
+
+Each entry in the binders list is computed as an HMAC over the portion
+of the ClientHello up to and including the PreSharedKeyExtension.identities
+field. That is, it includes all of the ClientHello but not the binders
+list itself.
```

Comment:
And this answers my question about why it's a separate list of binders with matching index, as opposed to a list of structures with identity and binder together.